server/src/main/java/org/elasticsearch/health/HealthService.java

```diff
@@ -11,25 +11,42 @@
 import org.elasticsearch.ResourceNotFoundException;
 import org.elasticsearch.core.Nullable;
 
+import java.util.Collections;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Locale;
+import java.util.Map;
 import java.util.Set;
 import java.util.TreeMap;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
+import static java.util.function.Predicate.isEqual;
 import static java.util.stream.Collectors.collectingAndThen;
 import static java.util.stream.Collectors.groupingBy;
 import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toMap;
 
 /**
  * This service collects health indicators from all modules and plugins of elasticsearch
  */
 public class HealthService {
 
+    // Visible for testing
+    static final String UNKNOWN_RESULT_SUMMARY = "Could not determine indicator state";
+
+    private final List<HealthIndicatorService> preflightHealthIndicatorServices;
     private final List<HealthIndicatorService> healthIndicatorServices;
 
-    public HealthService(List<HealthIndicatorService> healthIndicatorServices) {
+    /**
+     * @param preflightHealthIndicatorServices indicators that are run first and block health calculation until GREEN.
```

Comment:
Can we be more verbose on the documentation here - I don't think these indicators "block" the health calculation per se but rather indicate a severe problem in the system (?)